src/main/java/org/jenkinsci/plugins/durabletask/WindowsBatchScript.java

```diff
@@ -60,17 +60,18 @@ public String getScript() {
         BatchController c = new BatchController(ws);
 
         String cmd;
+        String quotedResultFile = quote(c.getResultFile(ws));
         if (capturingOutput) {
-            cmd = String.format("@echo off \r\ncmd /c \"\"%s\"\" > \"%s\" 2> \"%s\"\r\necho %%ERRORLEVEL%% > \"%s\"\r\n",
+            cmd = String.format("@echo off \r\ncmd /c \"\"%s\"\" > \"%s\" 2> \"%s\"\r\necho %%ERRORLEVEL%% > \"%s.tmp\"\r\nmove \"%s.tmp\" \"%s\"\r\n",
```

Comment:
And PowerShell?

Reply:
Powershell is caught by the 0-length check -- I have no idea how we'd even begin applying this to it and was planning to ask @gabloe or @jtnord because they know the Windows side of things. 

Can at least still write a Batch script though, so I included that.

Reply:
Sensible.